src/core/src/UsdUtil.cpp

```diff
@@ -222,25 +196,18 @@ pxr::SdfAssetPath getDynamicTextureProviderAssetPath(const std::string& name) {
     return pxr::SdfAssetPath(fmt::format("{}{}", rtx::resourcemanager::kDynamicTexturePrefix, name));
 }
 
-glm::dmat4 computeUsdToEcefTransform(const CesiumGeospatial::Cartographic& origin) {
-    return getEastNorthUpToFixedFrame(origin) * getAxisConversionTransform() * getUnitConversionTransform();
-}
-
-glm::dmat4 computeEcefToUsdTransform(const CesiumGeospatial::Cartographic& origin) {
-    return glm::inverse(computeUsdToEcefTransform(origin));
-}
-
 glm::dmat4
 computeEcefToUsdTransformForPrim(const CesiumGeospatial::Cartographic& origin, const pxr::SdfPath& primPath) {
-    const auto ecefToUsdTransform = computeEcefToUsdTransform(origin);
-    const auto primUsdWorldTransform = computeUsdWorldTransform(primPath);
-    const auto primEcefToUsdTransform = primUsdWorldTransform * ecefToUsdTransform;
+    const auto ecefToUsdTransform =
+        GeospatialUtil::getCoordinateSystem(origin, getUsdMetersPerUnit()).getEcefToLocalTransformation();
+    const auto primInverseUsdWorldTransform = glm::affineInverse(computeUsdWorldTransform(primPath));
+    const auto primEcefToUsdTransform = primInverseUsdWorldTransform * ecefToUsdTransform;
     return primEcefToUsdTransform;
 }
 
 glm::dmat4
 computeUsdToEcefTransformForPrim(const CesiumGeospatial::Cartographic& origin, const pxr::SdfPath& primPath) {
-    return glm::inverse(computeEcefToUsdTransformForPrim(origin, primPath));
+    return glm::affineInverse(computeEcefToUsdTransformForPrim(origin, primPath));
```

Comment:
Is this an optimization? I see `glm::inverse` in a few other places and wonder if we should use `affineInverse` exclusively. I don't see why we'd ever have any non-affine transformations.

Reply:
Yeah, I agree. As far as I'm able to tell, `cesium-native` prefers `affineInverse` as well. Seems like a safe optimization to me.

Reply:
Also, just to add... This exploits that the matrix is affine to be much faster in its calculation, although I admit I don't have any numbers handy.

Reply:
`inverse` vs. `affineInverse` could matter if we end up supporting non uniform scale, like stretching the tileset in the up-axis.

I haven't tried Unity or Unreal to see if it's expected to work in native.

Reply:
Never mind, an affine matrix can have non-uniform scale.